src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2358,6 +2386,7 @@ void *instanceservicesession(void *dummyPt) {
         }
     }
     instance_logger.log("Closing thread " + to_string(pthread_self()), "info");
+    pthread_exit(NULL);
     close(connFd);
```

Comment:
```suggestion
    close(connFd);
    pthread_exit(NULL);
```
Otherwise, close() will never be called.